### PR TITLE
Publishing

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -128,14 +146,49 @@ subprojects {
...
+    }
 }
 
-task removeProtoNames << {
-    File propsFile = file(getProtoPropertiesFilePath())
-    if (propsFile.exists()) {
-        propsFile.delete();
+task readPublishingCredentials << {
+    if (repositoryUserName == null || repositoryUserPassword == null) {
+        throw new InvalidUserDataException("Please set up valid credentials.")
```
Comment (alexander-yevsyukov) [inline build.gradle]: Please add description with the name of the file and its placement into the exception message text.

File: core-java-sample/build.gradle [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
 dependencies {
     testCompile group: 'junit', name: 'junit', version: '4.12'
 
+//    compile files('lib/core-java-0.1.jar') //works
+
+//    compile group: 'org.spine3', name: 'core-java', version: '0.1' //should work
+
     compile project(':core-java')
 }
```
Comment (alexander-yevsyukov) [inline core-java-sample/build.gradle]: GitHub complains about the absence of new line at the end of file.

File: core-java/build.gradle [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+//    repositories {
+//        maven {
+//            credentials {
+//                username 'mikhail.mikhaylov'
+//                password 'zxD123awdq2e'
+//            }
+//            url "http://maven.teamdev.com/repository/teamdev-maven"
+//        }
+//    }
+//}
```
Comment (alexander-yevsyukov) [inline core-java/build.gradle]: Do we really need these comments?

File: .gitignore [hunk 4]
```diff
@@ -3,4 +3,5 @@
 .idea/libraries
 .idea/*.xml
 */src/generated
-core-java/src/res/protos.properties
\ No newline at end of file
+core-java/src/res/protos.properties
+credentials.properties
```
Comment (alexander-yevsyukov) [inline .gitignore]: GitHub complains about the absence of new line at the end of file.

File: core-java-sample/build.gradle (hunk 2 above)
Comment (mikhailmikhaylov) [inline core-java-sample/build.gradle]: Fixed

File: core-java/build.gradle (hunk 3 above)
Comment (mikhailmikhaylov) [inline core-java/build.gradle]: Fixed

File: .gitignore (hunk 4 above)
Comment (mikhailmikhaylov) [inline .gitignore]: Fixed

File: build.gradle (hunk 1 above)
Comment (mikhailmikhaylov) [inline build.gradle]: Fixed

**Comment by mikhailmikhaylov** [comment]
Can we merge now?

**Pull request merged**